proposals/0072-feature-gate-threshold-automation.md

```diff
@@ -0,0 +1,241 @@
+  that returns the current epoch stake for a given vote account address.
+
+## Detailed Design
+
+The new process is comprised of the following steps:
+
+1. **Feature Creation:** Contributors create feature accounts as they do now.
+2. **Staging Features for Activation:** In some epoch N, a multi-signature
+   authority stages for activation some or all of the features created in step
+   1, to be activated at the end of the *next epoch*.
+3. **Signaling Support for Staged Features:** During the next epoch (epoch N+1),
+   validators signal which of the staged feature-gates they support in their
+   software.
+4. **Feature Activation:** At the end of epoch N+1, the runtime activates the
+   feature-gates that have the required stake support.
+
+### Step 1: Feature Creation
+
+The first step is creation of a feature account, done by submitting a
+transaction containing System instructions to fund, allocate, and assign the
+feature account to `Feature111111111111111111111111111111111111`.
+
+This step is unchanged from its original procedure.
+
+### Step 2: Staging Features for Activation
+
+A multi-signature authority, comprised of key-holders from Solana Labs and
+possibly other validator client teams in the future, will have the authority to
+stage created features for activation.
+In the future, this authority could be replaced by validator governance.
+
+The multi-signature authority stages a feature for activation by invoking the
+Feature Gate program's `StageFeatureForActivation` instruction. This instruction
+expects:
+
+- Data: The feature ID
+- Accounts:
+  - Staged Features PDA: writable
+  - Multi-signature authority: signer 
+
+One single PDA will be used to store two lists of features. Its data will be
+structured as follows:
+
+```c
+#define FEATURE_ID_SIZE 32
+#define MAX_FEATURES 8
+
+/**
+ * A Feature ID and its corresponding stake support, as signalled by validators.
+ */
+typedef struct {
+    /** Feature identifier (32 bytes). */
+    uint8_t feature_id[FEATURE_ID_SIZE];
+    /** Stake support (little-endian u64). */
+    uint8_t stake[8];
+} FeatureStake;
+
+/**
+ * Staged features for activation. 
+ */
+typedef struct {
+    /** The current epoch (little-endian u64). */
+    uint8_t current_epoch[8];
+    /**
+     * Features staged for activation at the end of the current epoch, with
+     * their corresponding signalled stake support.
+     */
+    FeatureStake current_feature_stakes[MAX_FEATURES];
+
+    /** The next epoch (little-endian u64). */
+    uint8_t next_epoch[8];
+    /** Feature IDs staged for the _next_ epoch. */
+    uint8_t next_features[MAX_FEATURES][FEATURE_ID_SIZE];
+} StagedFeatures;
+```
+
+`StageFeatureForActivation` will add the provided feature ID to the **next
+epoch's** set of staged features.
+
+An example of how this account's state might work to coordinate both lists can
+be found here:
+
+<https://github.com/buffalojoec/feature-gate-data-poc/blob/master/src/lib.rs>
```

Comment:
Remove, especially since it looks like the POC is out of date. If there is anything important in the POC implementation, please call it out in prose.